x/magpie/internal/keeper/handler.go

```diff
@@ -82,7 +82,13 @@ func handleMsgCreateSession(ctx sdk.Context, keeper Keeper, msg types.MsgCreateS
 		Signature:     msg.Signature,
 	}
 
-	if err := keeper.CreateSession(ctx, session); err != nil {
+	// Check for any previously existing session
+	if _, found := keeper.GetSession(ctx, session.SessionID); found {
+		return sdk.ErrUnknownRequest(fmt.Sprintf("Session with id %s already exists", session.SessionID)).Result()
```

Comment:
Not a big deal but looks strange if a function is named `GetSession` where the function is to check whether the session exists.

Reply:
The `GetSession` here is used to check if a session already exists, but elsewhere (i.e inside the `querier`) the same method is used to get a session using its ID.

Here we use the second returned value, which allows for faster and easier comparison than the first one. It's just a convinient way to keep the code DRY and avoiding having multiple methods with very similar functionality